lib_random/api/random_prng.h

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017, XMOS Ltd, All rights reserved
+#ifndef __RANDOM_PRNG_H__
+#define __RANDOM_PRNG_H__
+
+#include <stdint.h>
+#include <stddef.h>
+#include "random_impl.h"
+#include "random_pool.h"
+
+
+/** Interface for a client to request pseudo-random numbers. */
+interface random_prng {
+
+/** Method that returns a pseudo-random number.
+ *
+ *  ====== WARNING ====================================================
```

Comment:
@robert-lytton does this warning still hold true?
I'm trying to ensure I have correctly understood which applications this PRNG can be safely used for, but I'm struggling to reconcile the warning in the source with the comments in this review:

> >I will need to confirm that we will not need a CSPRNG for this

> The PRNG is revealing 32bits out of either 57, 88 or 113 bits, the other bits remain hidden (unless the user connects a debugger).

>It is up to the user to decide:
how many hidden bits are required;
how much entropy the seeds must start with;
how often entropy needs to be injected into the state;

>I believe this is best left as an explicit implementation detail.